src/app/shared/components/course-form/course-form.component.ts

```diff
@@ -28,17 +30,38 @@ export class CourseFormComponent implements OnInit {
   authorsListError!: boolean;
   authorsListArray!: FormArray;
   courseAuthorsListArray!: FormArray;
+  newCourse!: newCourse;
```

Comment:
Hmmm.. I'm not sure about the `newCourse` type. This is the same as the `Course` but having no id generated yet. Can you please refactor this to make it work with the existing `Course` model? Stick with a single object type.